frontend/src/store/reducers/taskReducer.js

```diff
@@ -0,0 +1,34 @@
+import { SET_TASKS, ADD_TASK, UPDATE_TASK, DELETE_TASK } from '../actions/types';
+
+const initialState = {
+    tasks: []
+};
+
+export default (state = initialState, { type, tasks, newTask, updatedTask, taskId }) => {
```

Comment:
leave only action here, don't destructure all possible fields, if you want to do that return from case a function where you pass state and action and in that function you can destructure relevant fields

Reply:
done